### DISCUSSION TITLE
Can Vercel Cron Jobs timeout?

### DISCUSSION THREAD

**Discussion by devon66h** [vercel/community#3302] [Feature Question] [answered]
### Question

Hi there, I see that Vercel functions [timeout after 10 sec.](https://vercel.com/guides/what-can-i-do-about-vercel-serverless-functions-timing-out#check-the-response) on the Hobby plan. Is this the same for Vercel Cron Jobs?

**Comment by amyegan** [comment, marked as answer]
Hey @devon66h. Yes, functions run by a cron job have the same timeout limits.

**Reply by karar-shah** [reply]
@amyegan I have a Vercel cron job through which I intend to send scheduled email reports created by GPT, summaries of content specific to a user in the database on the fly because the DB data get updated frequently. The issue lies in the cron job's timeout constraints. What is the most effective approach to achieve this functionality?

**Reply by amyegan** [reply]
Any AI generated content tends to take relatively long time to receive, so timeouts are common. You could try increasing the [max duration](https://vercel.com/docs/functions/runtimes#max-duration) for the function. Another option is to split it up so the report is generated in one step and the email is sent in second step. [Streaming](https://vercel.com/docs/functions/streaming#streaming-data-on-vercel) could help with the report generation step.

[a comment by an automated account is not shown]
